server/WFCORE-4251_list_modules_in_deployment_classpath.adoc

```diff
@@ -0,0 +1,118 @@
+* The user could use this information to discover any duplicated artifact bundled together with the deployment and included in one of the modules that the server is adding to the application classpath.
+* This information could facilitate discovering unneeded subsystems in the installation of his servers.
+
+
+The proposed solution is to include a new operation `list-modules` available under `/deployment` and, in case of ears, `/deployment=*/subdeployment` resource:
+
+ [standalone@localhost:9990 /] /deployment=ejb-in-ear.ear:list-modules
+
+ [standalone@localhost:9990 /] /deployment=ejb-in-ear.ear/subdeployment=ejb-in-ear-web.war:list-modules
+
+An example of the information returned by this method could be:
+
+  {
+      "outcome" => "success",
+      "result" => {
+          "ejb-in-ear.ear" => [
+              {
+                  "name" => "com.fasterxml.jackson.datatype.jackson-datatype-jdk8",
+                  "slot" => "main",
+                  "optional" => true,
+                  "export" => false,
+                  "import-services" => true,
+                  "duplicated" => false
+              },
+              {
+                  "name" => "com.fasterxml.jackson.datatype.jackson-datatype-jsr310",
+                  "slot" => "main",
+                  "optional" => true,
+                  "export" => false,
+                  "import-services" => true,
+                  "duplicated" => true
+              },
+              ...
+          ]
+      }
+  }
+
+Where:
+
+* name: The module name.
+* slot: The slot where this module is.
+* optional: If the dependency was added as an optional dependency.
+* export: If the dependency is being exported to other modules.
+* import-services: If this module is importing services from other dependencies.
+* duplicated: If it is a module defined by the user and the server is adding a module implicitly with the same name.
+
+
+== Issue Metadata
+
+=== Issue
+
+* {issue-base-url}WFCORE-4251[WFCORE-4251]
+
+=== Related Issues
+
+* {issue-base-url}EAP7-521[EAP7-521]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:szhantem@redhat.com[Sultan Zhantemirov]
+
+=== Affected Projects or Components
+
+* Wildfly Core
+* JBoss Modules (optional, it could help if we finally show the artifacts included on each module, see Nice-to-Have requirements)
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* It must be possible to list the implicit modules added to the classpath for a specific deployment and, in case of ears, for a particular sub-deployment.
+* List of modules should be sorted alphabetically by module name.
+* The list of modules will be only available if the deployment is enabled.
+* The new operation will be available only in standalone mode. Domain mode changes are not going to be covered by this RFE.
```

Comment:
I agree that there shouldn't be anything done on the HCs for this, but I don't see why this should not work for a domain mode server. The operation just has to addressed to /host=x/server=y/...

I suspect it would be more work to have it not work on a domain mode server than it would be to have it work.